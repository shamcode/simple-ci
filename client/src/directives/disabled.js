export default class {
    constructor() {
        this.node = null;
    }

    bind( node ) {
        this.node = node;
    }

    unbind() {
        this.node = null;
    }

    update( value ) {
        this.node.disabled = value;
    }
}
