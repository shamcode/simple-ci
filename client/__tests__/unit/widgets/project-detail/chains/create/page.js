import { DI } from 'sham-ui';
import Page from '../../../../../../src/widgets/project-detail/chains/create/page';
import directives from 'sham-ui-directives';
import renderer from 'sham-ui-test-helpers';

const flushPromises = () => new Promise( resolve => setImmediate( resolve ) );

afterEach( () => {
    DI.bind( 'store', null );
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    const storeMock = jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) );
    DI.bind( 'store', {
        getProjectById: storeMock
    } );
    DI.bind( 'router', {
        lastRouteResolved: jest.fn().mockReturnValueOnce( { params: { id: 1 } } ),
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( Page, {
        directives
    } );
    expect( storeMock.mock.calls[ 0 ][ 0 ] ).toBe( 1 );
    expect( meta.toJSON() ).toMatchSnapshot();
} );


it( 'create fail', async() => {
    expect.assertions( 2 );

    const createProjectChain = jest.fn();
    DI.bind( 'store', {
        getProjectById: jest.fn().mockReturnValue( Promise.resolve( {
            project: {
                id: 1,
                name: 'Test',
                cwd: '/tmp/'
            }
        } ) ),
        createProjectChain: createProjectChain
    } );
    DI.bind( 'router', {
        lastRouteResolved: jest.fn().mockReturnValueOnce( { params: { id: 1 } } ),
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( Page, {
        directives
    } );
    const { widget: { container } } = meta;
    await flushPromises();

    const formData = {
        name: 'Test name'
    };
    container.querySelector( '[name="name"]' ).value = formData.name;

    createProjectChain.mockReturnValueOnce( Promise.reject() );
    container.querySelector( '[type="submit"]' ).click();
    expect( createProjectChain.mock.calls ).toHaveLength( 1 );

    await flushPromises();
    expect( meta.toJSON() ).toMatchSnapshot();
} );
