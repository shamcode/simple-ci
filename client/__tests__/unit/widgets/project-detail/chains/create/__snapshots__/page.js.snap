// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create fail 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
  },
  "html": "
<ul class=\\"breadcrumb\\">
  <li>
    <a href=\\"/\\" class=\\"\\">Projects</a>
  </li>
  <li>
    <a href=\\"undefined\\" class=\\"\\">Test</a>
  </li>
  <!--if-->
  <!--if-->
  <li>Create project chain</li>
</ul>
<!--BreadCrumb-->
<ul class=\\"errors-panel\\">
  <li>Create project fail!</li>
</ul>
<!--if-->
<!--ErrorsPanel-->
<div class=\\"project-chain-form\\">
  <form>
    <div class=\\"group\\">
      <input name=\\"name\\" type=\\"text\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
      <label>Name</label>
    </div>
    <div class=\\"group\\">
      <textarea wrap=\\"off\\" style=\\"display: none\\" name=\\"command\\"></textarea>
      <div class=\\"code-flask\\">
        <div class=\\"codeflask codeflask--has-line-numbers\\">
          <textarea class=\\"codeflask__textarea codeflask__flatten\\" spellcheck=\\"false\\" autocapitalize=\\"off\\" autocomplete=\\"off\\" autocorrect=\\"off\\"></textarea><pre class=\\"codeflask__pre codeflask__flatten  language-html\\"><code class=\\"codeflask__code  language-html\\"></code></pre>
          <div class=\\"codeflask__lines\\"><span class=\\"codeflask__lines__line\\">1</span></div>
        </div>
      </div>
      <!--CodeFlask--><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
      <label>Command</label>
    </div>
    <div class=\\"group\\">
      <button type=\\"submit\\"><span>Create</span></button>
    </div>
  </form>
</div>
<!--ChainForm-->
<!--if-->
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "afterRender": [Function],
    "dataLoaded": true,
    "dataSaving": false,
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
    "errors": Array [
      "Create project fail!",
    ],
    "project": Object {
      "cwd": "/tmp/",
      "id": 1,
      "name": "Test",
    },
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
  },
  "html": "
<ul class=\\"breadcrumb\\">
  <li>
    <a href=\\"/\\" class=\\"\\">Projects</a>
  </li>
  <!--if-->
  <!--if-->
  <li></li>
</ul>
<!--BreadCrumb-->
<!--if-->
<!--FlushPageDataLoading-->
<!--if-->
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "afterRender": [Function],
    "dataLoaded": false,
    "dataSaving": false,
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
    "errors": Array [],
    "project": Object {},
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
